### PR TITLE
Fix CoreLib fast up-to-date check

### LINKED ISSUE #68698: System.Private.CoreLib's fast up-to-date check is broken
When building System.Runtime's src project, which depends on System.Private.CoreLib without making any changes to any of the assemblies, Systme.Private.CoreLib always incrementally builds:

```
1>------ Build started: Project: System.Private.CoreLib, Configuration: Debug x64 ------
1>System.Private.CoreLib -> C:\git\runtime5\artifacts\bin\coreclr\windows.x64.Debug\IL\System.Private.CoreLib.dll
========== Build: 1 succeeded, 0 failed, 5 up-to-date, 0 skipped ==========

1>------ Build started: Project: System.Private.CoreLib, Configuration: Debug x64 ------
1>System.Private.CoreLib -> C:\git\runtime5\artifacts\bin\coreclr\windows.x64.Debug\IL\System.Private.CoreLib.dll
========== Build: 1 succeeded, 0 failed, 5 up-to-date, 0 skipped ==========

1>------ Build started: Project: System.Private.CoreLib, Configuration: Debug x64 ------
1>System.Private.CoreLib -> C:\git\runtime5\artifacts\bin\coreclr\windows.x64.Debug\IL\System.Private.CoreLib.dll
========== Build: 1 succeeded, 0 failed, 5 up-to-date, 0 skipped ==========
```

This shouldn't happen as Visual Studio caches inputs and outputs and makes sure that a project's build isn't even triggered when nothing changed.

cc @dotnet/runtime-infrastructure @trylek @agocke @jkotas

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes https://github.com/dotnet/runtime/issues/68698

CoreLib's fast up-to-date check was broken because the output pdb was
moved into a different folder. Changing this to use a copy task instead
and consolidate the two targets into one.

I couldn't find a publicly exposed property that allows to change the
pdb's output path, hence I decided to copy.

### CODE REVIEW

File: eng/NoTargetsSdk.BeforeTargets.targets [hunk 1]
```diff
@@ -5,4 +5,4 @@
     <TargetFramework Condition="'$(TargetFramework)' == '' and '$(TargetFrameworks)' == ''">$(NetCoreAppCurrent)</TargetFramework>
   </PropertyGroup>
 
-</Project>
\ No newline at end of file
+</Project>
```
Comment (ViktorHofer) [inline eng/NoTargetsSdk.BeforeTargets.targets:8]: Please just ignore this change. I originally had a change in that file and then reverted it in the GH UI. Don't want to spin another build for this...

File: src/coreclr/Directory.Build.targets [hunk 2]
```diff
@@ -2,36 +2,28 @@
...
-    <Move SourceFiles="@(_AllPdbs)"
-          DestinationFolder="$(RuntimeBinDir)PDB/aotsdk"
-          Condition="'%(_AllPdbs.FolderName)' == 'aotsdk'" />
-
+          AfterTargets="CopyFilesToOutputDirectory"
+          DependsOnTargets="_CheckForCompileOutputs"
+          Inputs="@(_DebugSymbolToMove)"
+          Outputs="@(_DebugSymbolToMove->Metadata('Destination'))"
+          Condition="'$(_DebugSymbolsProduced)' == 'true'">
+    <Copy SourceFiles="@(_DebugSymbolToMove)"
```
Comment (agocke) [inline src/coreclr/Directory.Build.targets:24]: Any particular reason we're avoiding `SkipUnchangedFiles`?

**Review by agocke — APPROVED** [review]
LGTM aside from small comment

File: src/coreclr/Directory.Build.targets (hunk 2 above)
Comment (ViktorHofer) [reply src/coreclr/Directory.Build.targets:24]: That setting isn't useful when you leverage Target Inputs and Outputs. The documentation actually warns on using it: https://docs.microsoft.com/en-us/visualstudio/msbuild/copy-task?view=vs-2022.

**Comment by ViktorHofer** [comment]
The two test failures are known and got already disabled. I pinged @jakobbotsch offline regarding the msbuild failure as I'm unsure if it's a known issue.

File: src/coreclr/Directory.Build.targets [hunk 3]
```diff
@@ -2,36 +2,28 @@
...
-          Condition="Exists(@(_AllPdbs))"
-          Inputs="@(_AllPdbs)"
-          Outputs="@(_AllPdbs -> '$(RuntimeBinDir)PDB/aotsdk/%(Filename).pdb')">
-
-    <Move SourceFiles="@(_AllPdbs)"
-          DestinationFolder="$(RuntimeBinDir)PDB/aotsdk"
-          Condition="'%(_AllPdbs.FolderName)' == 'aotsdk'" />
-
+          AfterTargets="CopyFilesToOutputDirectory"
+          DependsOnTargets="_CheckForCompileOutputs"
```
Comment (hoyosjs) [inline src/coreclr/Directory.Build.targets:20]: Is there any non-internal target that we can hang off of? Like this target is implied by `CopyFilesToOutputDirectory`. Is the AfterTargets here not enough?

File: src/coreclr/System.Private.CoreLib/System.Private.CoreLib.csproj [hunk 4]
```diff
@@ -9,7 +9,7 @@
     <TargetFramework>$(NetCoreAppCurrent)</TargetFramework>
 
     <!-- Force System.Private.CoreLib.dll into a special IL output directory -->
-    <OutputPath>$(RuntimeBinDir)/IL/</OutputPath>
+    <OutputPath>$(RuntimeBinDir)IL\</OutputPath>
```
Comment (hoyosjs) [inline src/coreclr/System.Private.CoreLib/System.Private.CoreLib.csproj:12]: Don't we usually normalize or use linux path separators?

**Pull request merged**